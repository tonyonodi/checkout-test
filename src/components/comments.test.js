import React from "react";
import renderer from "react-test-renderer";
import Comments from "./Comments";

it("renders comment correctly", () => {
  const tree = renderer
    .create(
      <Comments
        comments={[
          {
            name: "Alice",
            email: "alice@gmail.com",
            rating: 4,
            comment: "Some comment.",
            timeSubmitted: new Date("2019-02-26T11:39:26.259Z"),
            id: 1,
          },
          {
            name: "Bob",
            email: "bob@gmail.com",
            rating: 2,
            comment: "Another comment.",
            timeSubmitted: new Date("2019-02-26T11:39:35.916Z"),
            id: 2,
          },
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders empty comments correctly", () => {
  const tree = renderer.create(<Comments comments={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});
