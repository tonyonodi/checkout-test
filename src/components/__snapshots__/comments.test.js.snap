// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comment correctly 1`] = `
<div
  className="sc-EHOje jvzENE"
>
  <div
    className="sc-bdVaJa bDWFJH"
  >
    <h3
      className="sc-bwzfXH kfvjYX"
    >
      Alice
    </h3>
    <div
      className="sc-htpNat gHoJWV"
    >
      <span>
        ⭐
      </span>
      <span>
        ⭐
      </span>
      <span>
        ⭐
      </span>
      <span>
        ⭐
      </span>
    </div>
    <div
      className="sc-bxivhb bUZGRr"
    >
      <p>
        Some comment.
      </p>
    </div>
  </div>
  <div
    className="sc-bdVaJa bDWFJH"
  >
    <h3
      className="sc-bwzfXH kfvjYX"
    >
      Bob
    </h3>
    <div
      className="sc-htpNat gHoJWV"
    >
      <span>
        ⭐
      </span>
      <span>
        ⭐
      </span>
    </div>
    <div
      className="sc-bxivhb bUZGRr"
    >
      <p>
        Another comment.
      </p>
    </div>
  </div>
</div>
`;

exports[`renders empty comments correctly 1`] = `
<div
  className="sc-ifAKCX iwqfNv"
>
  There are no comments yet.
</div>
`;
